pyamg/strength.py

```diff
@@ -175,37 +182,81 @@ def classical_strength_of_connection(A, theta=0.0, norm='abs'):
     else:
         blocksize = 1
 
-    if not sparse.isspmatrix_csr(A):
-        warn('Implicit conversion of A to csr', sparse.SparseEfficiencyWarning)
-        A = sparse.csr_matrix(A)
-
     if (theta < 0 or theta > 1):
         raise ValueError('expected theta in [0,1]')
 
```

Comment:
I think there could be an error if the user chooses block='block', but A is CSR.  Then, I think the line

       R,C = A.blocksize

will fail.  How about we utilize the above code and do something like 

    if sparse.isspmatrix_bsr(A):
        blocksize = A.blocksize[0]
    else:
        blocksize = (1,1)

and then use only this blocksize in the code?

Reply:
think this wouldn't quite work because if a is BSR this give scalar. you're right though, pushed a fix where R==C check is done first in code, then just use scalar blocksize everywhere

Reply:
Sounds good!